localstack/services/s3/v3/models.py

```diff
@@ -0,0 +1,607 @@
+    etag: Optional[ETag]
+    user_metadata: Metadata
+    system_metadata: Metadata
+    last_modified: datetime
+    expires: Optional[datetime]
+    expiration: Optional[Expiration]  # come from lifecycle
+    storage_class: StorageClass | ObjectStorageClass
+    encryption: Optional[ServerSideEncryption]  # inherit bucket
+    kms_key_id: Optional[SSEKMSKeyId]  # inherit bucket
+    bucket_key_enabled: Optional[bool]  # inherit bucket
+    checksum_algorithm: ChecksumAlgorithm
+    checksum_value: str
+    lock_mode: Optional[ObjectLockMode]
+    lock_legal_status: Optional[ObjectLockLegalHoldStatus]
+    lock_until: Optional[datetime]
+    website_redirect_location: Optional[WebsiteRedirectLocation]
+    acl: Optional[str]  # TODO: we need to change something here, how it's done?
+    is_current: bool
+    parts: Optional[list[tuple[int, int]]]
+    restore: Optional[Restore]
+
+    def __init__(
+        self,
+        key: ObjectKey,
+        etag: Optional[ETag] = None,
+        size: Optional[int] = None,
+        version_id: Optional[ObjectVersionId] = None,
+        user_metadata: Optional[Metadata] = None,
+        system_metadata: Optional[Metadata] = None,
+        storage_class: StorageClass = StorageClass.STANDARD,
+        expires: Optional[datetime] = None,
+        expiration: Optional[Expiration] = None,
+        checksum_algorithm: Optional[ChecksumAlgorithm] = None,
+        checksum_value: Optional[str] = None,
+        encryption: Optional[ServerSideEncryption] = None,
+        kms_key_id: Optional[SSEKMSKeyId] = None,
+        bucket_key_enabled: bool = False,
+        lock_mode: Optional[ObjectLockMode] = None,
+        lock_legal_status: Optional[ObjectLockLegalHoldStatus] = None,
+        lock_until: Optional[datetime] = None,
+        website_redirect_location: Optional[WebsiteRedirectLocation] = None,
+        acl: Optional[str] = None,  # TODO
+    ):
+        self.key = key
+        self.user_metadata = (
+            {k.lower(): v for k, v in user_metadata.items()} if user_metadata else {}
+        )
+        self.system_metadata = system_metadata or {}
+        self.version_id = version_id
+        self.storage_class = storage_class or StorageClass.STANDARD
+        self.etag = etag
+        self.size = size
+        self.expires = expires
+        self.checksum_algorithm = checksum_algorithm
+        self.checksum_value = checksum_value
+        self.encryption = encryption
+        self.kms_key_id = kms_key_id
+        self.bucket_key_enabled = bucket_key_enabled
+        self.lock_mode = lock_mode
+        self.lock_legal_status = lock_legal_status
+        self.lock_until = lock_until
+        self.acl = acl
+        self.expiration = expiration
+        self.website_redirect_location = website_redirect_location
+        self.is_current = True
+        self.last_modified = datetime.utcnow()
+        self.parts = []
+        self.restore = None
+
+    def get_system_metadata_fields(self):
```

Comment:
I think these should return a dict, because they are then merged into the TypedDict response like `**`. Thanks for pointing it out, I will correct that `Headers` thing 😬